src/mcp23017.cpp

```diff
@@ -36,6 +99,20 @@ int Mcp23017::begin(uint8_t directions[8]) {
     int rc;
 
     // TODO: Add device ID check
+    Wire.requestFrom(this->addr, 1);
+    if (Wire.available()){
+        rc = Wire.read();
```

Comment:
You should add a check here to make sure the value you read from the register is actually the expected default value of the register. Also, don't forget to add the "write register address to the device" part that I mentioned in another comment.